### DISCUSSION TITLE
Tokio not utilizing full cpu on compute heavy task

### DISCUSSION THREAD

**Discussion by ktjd123** [tokio-rs/tokio#5779] [Q&A]
Hi,

I'm currently spawning about 50,000 tokio threads using `tokio::spawn` and joining all at end.

Each thread has relatively simple task, taking about 100ms ~ 1s

The problem is tokio seems to be not utilizing full cpu for concurrent job.

How can I solve this problem?

It seems code right after spawn executes after some time passed. -> I'm using loop for spawn so checked at end of loop that every spawning code have done executing `tokio::spawn`

I've checked with two machine and having same behavior. -> Utilizing about 20% (when inside task is slightly simple data) of cpu or 80% (when inside task is slightly complex with data)

1. Mac with apple silicon having 8cores.
2. Ubuntu with arm 80 cores.

Thank you in advance.

**Comment by carllerche** [comment]
Would you happen to have a repro or code examples? There could be many reasons; without more data, it is hard to say.

**Comment by WuBingzheng** [comment]
I got the same problem.
I post another [discussion](https://github.com/tokio-rs/tokio/discussions/5991) with source code.
